ecscollector/collector.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package ecscollector implements a Prometheus collector for Amazon ECS
+// metrics available at the ECS metadata server.
+package ecscollector
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/prometheus-community/ecs_exporter/ecsmetadata"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	numProcsDesc = prometheus.NewDesc(
+		"ecs_processes",
+		"Number of processes.",
+		labels, nil)
+
+	cpuTotalDesc = prometheus.NewDesc(
+		"ecs_cpu_total",
+		"Jiffies of total CPU usage.",
+		cpuLabels, nil)
```

Comment:
This needs to be converged to seconds. We can assume the scaling value is 100hz, so divide jiffies by 100 to get seconds.

Reply:
Jiffies are platform specific and we can read it from _SC_CLK_TCK, defaulting to 100 and making a sysconf call.